datalake/src/main/java/com/sequenceiq/datalake/service/recovery/SdxRecoverySelectorService.java

```diff
@@ -0,0 +1,69 @@
+package com.sequenceiq.datalake.service.recovery;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.dto.NameOrCrn;
+import com.sequenceiq.cloudbreak.common.exception.BadRequestException;
+import com.sequenceiq.sdx.api.model.SdxRecoverableResponse;
+import com.sequenceiq.sdx.api.model.SdxRecoveryRequest;
+import com.sequenceiq.sdx.api.model.SdxRecoveryResponse;
+
+@Service
+/**
+ * Recovers an SDX cluster from a failure during Resize or Upgrade.
+ *
+ * Chooses the appropriate recovery action to take based on the state of the SDX cluster. Examples:
+ * <ul>
+ *     <li>{@Code SdxUpgradeRecoveryService}</li>
+ *     <li>{@Code ResizeRecoveryService}</li>
+ * </ul>
+ */
+public class SdxRecoverySelectorService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SdxRecoverySelectorService.class);
+
+    private List<RecoveryService> services;
+
+    @Autowired
+    public SdxRecoverySelectorService(List<RecoveryService> services) {
+        this.services = services;
+    }
+
+    public SdxRecoveryResponse triggerRecovery(String crn, NameOrCrn nameOrCrn, SdxRecoveryRequest sdxRecoveryRequest) {
```

Comment:
crn should be renamed to userCrn everywhere. but as I mentioned in another comment, I am not sure if userCrn parameter is needed at all